neqo-transport/src/send_stream.rs

```diff
@@ -1128,10 +1132,10 @@ impl SendStream {
             SendStreamState::Send {
                 ref mut send_buf, ..
             } => {
+                let previous_limit = send_buf.avail();
                 send_buf.mark_as_acked(offset, len);
-                if self.avail() > 0 {
-                    self.conn_events.send_stream_writable(self.stream_id);
-                }
+                let current_limit = send_buf.avail();
```

Comment:
I am not sure I am following. The above `send_buf.mark_as_acked` only alters the `send_buf` limit. Not the connection or stream flow control limit.

> but the real check here is whether the peer has given out enough flow control credit.

In case the peer has not given out enough flow control credit, `maybe_emit_writable_event` will not emit a writable event, due to the block below in the function:

``` rust
        if self.avail() < low_watermark {
            // Stream is constrained by different limit.
            return;
        }
```

The case where the peer gives more flow control credit, and thus potentially a `Writable` event is due, is handled in `set_max_stream_data`.

@martinthomson what am I missing?

Reply:
Ahh, so the checks already exists.  This is fine.  Disregard.